Revamp Hive image build process (#581)
stackabletech/docker-images: README.adoc

This repository contains Dockerfiles and scripts to build base images for use within Stackable.

== Prerequisites

* Stackable Image Tools (`pip install image-tools-stackabletech`) https://github.com/stackabletech/image-tools
* Docker including the `buildx` plugin: https://github.com/docker/buildx
* Optional: https://github.com/redhat-openshift-ecosystem/openshift-preflight[OpenShift preflight tool] to verify an image for OpenShift

== Build Product Images

Product images are published to the `docker.stackable.tech` registry under the `stackable` organization by default.

> To build and push product images to the default repository, use the `build_product_images.py` like this:

Reviewer: ```suggestion
The [image-tools](https://github.com/stackabletech/image-tools) Python package provides a command line program called `bake` that can be used for this purpose.

To build and push product images to the default repository:
```

>     bake --product zookeeper --image 0.0.0-dev --push

Reviewer 2: Maybe it's just me, but this is a little confusing.

It says to use `build_product_images.py`, but goes on to use `bake`.

Reviewer 2: I see this section was moved, so nothing technically _needs_ fixing in this PR.

Author: I am confused by this as well though. I'll ping @razvan - maybe he can help

Reviewer: It's an omission. The py script doesn't exist anymore

Author: I removed the reference.

Reviewer 2: f405a3c

This will build images for Apache ZooKeeper versions as defined in the `conf.py` file, tag them with the `image-version` 0.0.0-dev and push them to the registry.

> You can select a specific version of a product to build using the syntax `product=version` e.g. to build Hive 3.1.3 you can use this command:

Reviewer 2: ```suggestion
You can build a specific version of a product by using the syntax `product=version` e.g. to build Hive 3.1.3 you can use this command:
```

    bake --product hive=3.1.3 -i 0.0.0-dev

> NOTE: `-i` is the shorthand for `--image`

Reviewer 2: I'm not very familiar with `bake`, but is `-i` for just the image tag, or can you specify a whole new tag?

Eg: could I do `-i my.registry/image:dev`

```suggestion
NOTE: `-i` is the shorthand for `--image` (ie: the resulting image tag)
```

Author: Fixed in [`6e0cf20` (#581)](https://github.com/stackabletech/docker-images/pull/581/commits/6e0cf20eb078c394cc02a175ab5277c31b5fec8d)

Author: To answer your question: I have no idea myself. I'm just changing existing documentation here :(

Reviewer: If only there were a --help option:
